src/lib/hooks/useModalSelect.ts

```diff
@@ -0,0 +1,35 @@
+//- Hook Imports
+import { useModal } from './useModal';
+
+//- Constants Imports
+import { ModalType } from '../constants/modals';
+
+type UseModalSelectReturn = {
+	handleModal: () => void;
+};
+
+export const useModalSelect = (accountId?: string): UseModalSelectReturn => {
+	const { openModal, closeModal } = useModal();
+
+	// todo: handle modal will need expanding when modal content is added
+	const handleModal = (domainName?: string, type?: ModalType) => {
+		type &&
+			accountId &&
+			openModal({
+				modalType: type,
+				contentProps: {
+					domainName: domainName,
+					onClose: closeModal,
+				},
+			});
```

Comment:
nit: I think this could be cleaner if `openModal` had the signature `openModal(modal: Modal, props: any)`. That way it would look like `openModal(Modal.Bid, { domainName: 'wilder.wheels' })`.

The reason for this is it will reduce nesting by one level, and in my opinion it makes the function more obvious - you pass it a modal and some props to apply to it, rather than defining some object keys.

Not sure what that prop type would be but there's probably some typescript magic you can use to get it match whatever modal you're rendering.

Reply:
Over the weekend I had a look at how you were handling modals in zApp-Staking and decided to follow this approach to reduce complexity. Each modal is now nested in its own feature, which can be controlled using a button component.

removed useModalSelect due to the above changes.